airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java

```diff
@@ -63,23 +64,74 @@ public class TemporalClient {
-    final WorkflowServiceStubs temporalService = TemporalUtils.createTemporalService(temporalHost);
-    return new TemporalClient(WorkflowClient.newInstance(temporalService), workspaceRoot, temporalService, configs);
+  public static TemporalClient cloud(final Configs configs) {
+    log.info("Using Temporal Cloud with:\nhost: {}\nnamespace: {}", configs.getTemporalCloudHost(), configs.getTemporalCloudNamespace());
+    final WorkflowServiceStubs temporalCloudService = TemporalUtils.createTemporalCloudService();
+    return new TemporalClient(
+        WorkflowClient.newInstance(
+            temporalCloudService,
+            WorkflowClientOptions.newBuilder()
+                .setNamespace(configs.getTemporalCloudNamespace())
+                .build()),
+        configs.getWorkspaceRoot(),
+        temporalCloudService);
+  }
+
+  public static TemporalClient airbyte(final Configs configs) {
+    final String temporalHost = configs.getTemporalHost();
+    log.info("Using Temporal Airbyte with:\nhost: {}\nnamespace: {}", temporalHost, TemporalUtils.DEFAULT_NAMESPACE);
+    final WorkflowServiceStubs temporalService = TemporalUtils.createTemporalAirbyteService(temporalHost);
+    return new TemporalClient(WorkflowClient.newInstance(temporalService), configs.getWorkspaceRoot(), temporalService);
+  }
+
+  private static WorkflowClient cloudWorkflowClient(final WorkflowServiceStubs temporalService, final Configs configs) {
+    return WorkflowClient.newInstance(
+        temporalService,
+        WorkflowClientOptions.newBuilder()
+            .setNamespace(configs.getTemporalCloudNamespace())
+            .build());
+  }
+
+  private static WorkflowClient airbyteWorkflowClient(final WorkflowServiceStubs temporalService) {
+    return WorkflowClient.newInstance(temporalService);
   }
 
   // todo (cgardens) - there are two sources of truth on workspace root. we need to get this down to
   // one. either temporal decides and can report it or it is injected into temporal runs.
-  public TemporalClient(final WorkflowClient client,
-                        final Path workspaceRoot,
-                        final WorkflowServiceStubs workflowServiceStubs,
-                        final Configs configs) {
+  @VisibleForTesting
+  protected TemporalClient(final WorkflowClient client,
+                           final Path workspaceRoot,
+                           final WorkflowServiceStubs workflowServiceStubs) {
     this.client = client;
     this.workspaceRoot = workspaceRoot;
     this.service = workflowServiceStubs;
-    this.configs = configs;
+  }
+
+  /**
+   * Direct termination of Temporal Workflows should generally be avoided. This method exists for some
+   * rare circumstances where this may be required. Originally added to facilitate Airbyte's migration
+   * to Temporal Cloud.
+   */
+  public void dangerouslyTerminateWorkflow(final String workflowId, final String reason) {
```

Comment:
The migrator app calls this to terminate workflows that it is migrating from airbyte-temporal to Temporal Cloud